tsl/test/shared/sql/constify_timestamptz_op_interval.sql.in

```diff
@@ -0,0 +1,48 @@
+-- This file and its contents are licensed under the Timescale License.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-TIMESCALE for a copy of the license.
+
+\set PREFIX 'EXPLAIN (costs off)'
+
+-- we disable ChunkAppend and ConstraintAwareAppend here to make the exclusion easier to spot
+-- otherwise those would remove the chunks from the plan during execution
+SET timescaledb.enable_chunk_append TO false;
+SET timescaledb.enable_constraint_aware_append TO false;
+
+-- plan query on complete hypertable to get a list of the chunks
+:PREFIX SELECT time FROM metrics;
+
+-- all of these should have all chunk exclusion happening at plan time
+:PREFIX SELECT time FROM metrics WHERE time < '2000-01-01'::timestamptz - '6h'::interval;
+:PREFIX SELECT time FROM metrics WHERE time < '2000-01-01'::timestamptz + '6h'::interval;
+:PREFIX SELECT time FROM metrics WHERE time < '6h'::interval + '2000-01-01'::timestamptz;
+:PREFIX SELECT time FROM metrics WHERE time < '2000-01-07'::timestamptz - '7 day 8 seconds'::interval;
+:PREFIX SELECT time FROM metrics WHERE time < '2000-03-01'::timestamptz - '60 day'::interval;
+
+-- test Var on right side of expression
+:PREFIX SELECT time FROM metrics WHERE '2000-01-01'::timestamptz - '6h'::interval > time;
+:PREFIX SELECT time FROM metrics WHERE '2000-01-07'::timestamptz - '7 day'::interval > time;
+:PREFIX SELECT time FROM metrics WHERE '2000-03-01'::timestamptz - '60 day'::interval > time;
+
+-- test multiple constraints
+:PREFIX SELECT time FROM metrics WHERE time > '2000-01-10'::timestamptz - '6h'::interval AND time < '2000-01-10'::timestamptz + '6h'::interval;
+
+-- test on space-partitioned hypertable
+:PREFIX SELECT time FROM metrics_space WHERE time < '2000-01-01'::timestamptz - '6h'::interval AND device_id = 1;
+-- test on compressed hypertable
+:PREFIX SELECT time FROM metrics_compressed WHERE time < '2000-01-01'::timestamptz - '6h'::interval;
+-- test on space-partitioned compressed hypertable
+:PREFIX SELECT time FROM metrics_space_compressed WHERE time < '2000-01-01'::timestamptz - '6h'::interval AND device_id = 1;
+
+-- month/year intervals are not constified
+:PREFIX SELECT time FROM metrics WHERE time < '2000-02-01'::timestamptz - '1 month'::interval;
+:PREFIX SELECT time FROM metrics WHERE time < '2000-02-01'::timestamptz - '1 month - 1 day'::interval;
```

Comment:
Please add tests for month plus something else as well. If a change is made that for some reason check the non-month part first and accidentally skip checking the month, it would proceed without an error even though it shouldn't. Yes, I'm being paranoid, but I've seen these kind of errors more often than I like.

Reply:
i'll add but in the interval representation month is stored separately so '1 month - 1 day' will be {.month = 1, .day = -1 } days wont reduce the month value

Reply:
Yes, I've seen the code, but I just want to have tests for both positive and negative "day" parts because it sounds like something that developers can "abuse". E.g., "If it is negative, I know it is a month involved, otherwise it is not so...".